### PR TITLE
Prepare for v21.12

### PR DESCRIPTION
**Pull request by guparan**
This PR should be squashed (= merged with the "Squash and merge" button) and then cherry-picked into v21.12 branch.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-20.04, macos-10.15, windows-2019]
-        sofa_branch: [master]
+        sofa_branch: [master, v21.12]
```
Comment (EulalieCoevoet) [inline .github/workflows/ci.yml:16]: I've already added this in the branch v21.12 and I don't think we should have that on master right?
The goal is not to keep master working with this version.

File: cmake/environment.cmake [hunk 2]
```diff
@@ -1,17 +1,17 @@
 # CMake modules path, for our FindXXX.cmake modules
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
-list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
-list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR}/extlibs)
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
```
Comment (EulalieCoevoet) [inline cmake/environment.cmake:3]: The directory is still "modules" with lower case right?

Comment (guparan) [reply cmake/environment.cmake:3]: Indeed, bad copy/paste

File: .github/workflows/ci.yml (hunk 1 above)
Comment (guparan) [reply .github/workflows/ci.yml:16]: It's actually a good idea to keep it to know when master will not be compatible with v21.12 anymore (and to remove it at this moment).

**Review by EulalieCoevoet — APPROVED** [review]

**Pull request merged**